src/zenml/orchestrators/utils.py

```diff
@@ -233,11 +233,12 @@ def _link_cached_artifacts_to_model(
         if artifact_config_ is None:
             artifact_config_ = ArtifactConfig(name=output_name_)
 
-        link_artifact_config_to_model(
-            artifact_config=artifact_config_,
-            model=model_from_context,
-            artifact_version_id=output_id,
-        )
+        for output_id in output_ids:
+            link_artifact_config_to_model(
+                artifact_config=artifact_config_,
```

Comment:
The artifact config that is passed here is only valid for artifacts that were returned as the step output, not for dynamic ones saved using `save_artifact`. Are those already included in `step_run.outputs` at this point? And if yes, I would assume they're already linked right?